src/Rules.hs

```diff
@@ -39,6 +39,8 @@ buildLib stage pkg = do
     when (pkg `elem` activePackages) $
         if isLibrary pkg
         then do -- build a library
+            when (nonCabalContext context) $
```

Comment:
@izgzhen Did you mean `unless (nonCabalContext context)` here? I've just stubled upon this and have no idea why this works. Non-Cabal packages do not have `setup-config` files... I guess this was just a no-op!

Reply:
As I remember, I was guarding the `need` with this condition because `setup-config` somehow depends on `ghc-cabal`, so the later package can't depend on `setup-config`. I wrote this line based on empirical observation -- and it works somehow :$

Reply:
Thanks, I see. Anyway, the opposite condition also works :) Maybe it means we don't need it at all? I'll think about it and may end up deleting this bit, or moving it elsewhere.

If by doing this I accidentally break installation build rules, we'll know it's my fault!

Reply:
Just as a reference, here is the opposite condition: 

https://github.com/snowleopard/hadrian/blob/master/src/Rules.hs#L71